pkg/enforcement/check.go

```diff
@@ -30,31 +38,90 @@ func NewCheckRequest(user User, action Action, resource Resource, context map[st
 	}
 }
 
+func newJsonCheckRequest(opaUrl string, user User, action Action, resource Resource, context map[string]string) ([]byte, error) {
+	checkReq := NewCheckRequest(user, action, resource, context)
+	var jsonCheckReq []byte
+	var err error
+	if opaUrl != "" {
+		opaCheckReq := &opaCheckRequest{
+			checkReq,
+		}
+		jsonCheckReq, err = json.Marshal(opaCheckReq)
+	} else {
+		jsonCheckReq, err = json.Marshal(checkReq)
```

Comment:
Maybe you can drop one of the calls to Marshal by first setting an `interface{}` generic request variable (overriding it with the opa-specific version if needed) and then calling Marshal on that